src/main/java/scrolls/elder/logic/parser/FindCommandParser.java

```diff
@@ -25,9 +33,60 @@ public FindCommand parse(String args) throws ParseException {
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
 
-        String[] nameKeywords = trimmedArgs.split("\\s+");
+        FindType findType = parseForRoles(trimmedArgs.split("\\s+"));
+
+        String newArgs = args.replace("r/volunteer", "").replace("r/befriendee", "");
+        String newTrimmedArgs = newArgs.trim();
+
+        if (newTrimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        }
+
+        String[] newNameKeywords = newTrimmedArgs.split("\\s+");
+
+        if (findType.equals(FindType.BOTH)) {
+            return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(newNameKeywords)),
+                    true, true);
+        } else if (findType.equals(FindType.VOLUNTEER_ONLY)) {
+            return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(newNameKeywords)),
+                    true, false);
+        } else {
+            return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(newNameKeywords)),
+                    false, true);
+        }
+
+    }
+
+    private static FindType parseForRoles(String[] nameKeywords) throws ParseException {
+        boolean searchVolunteer = false;
+        boolean searchBefriendee = false;
```

Comment:
Could perhaps rename these boolean variables to be more boolean sounding

Reply:
Noted on this